.github/workflows/report-failure.yaml

```
@@ -0,0 +1,87 @@
+# This workflow is intended to be called by workflows in our various software
+# repos to have shared workflow failure reporting in the form of GitHub issues.
+on:
+  workflow_call:
+    inputs:
+      failure:
+        description: Whether to create an issue indicating a failed workflow run, or keep an existing issue open.
+        type: boolean
+        required: true
+      success:
+        description: Whether to close any open issues created by this workflow.
+        type: boolean
+        required: true
+      title:
+        description: Title for issues opened by this workflow.
+        type: string
+        required: true
+    secrets:
+      token:
+        description: A GitHub access token with permission to create and modify GitHub issues.
+        required: true
+
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Extract workflow link
+        run: |
+          workflow_path=$(echo "${{ github.workflow_ref }}" | sed -E 's|^.*/(\.github/workflows/[^@]+)@\S*$|\1|')
+          echo "workflow_link=https://github.com/${{ github.repository }}/blob/${{ github.sha }}/$workflow_path" >> "$GITHUB_ENV"
```

I strongly suggest we avoid shell escaping issues (e.g. `"` in `github.workflow_ref`) by using environment variables instead of interpolations during YAML preprocessing. The default environment is enough; no need for step-specific ones to hold the context vars.

```suggestion
          workflow_path=$(echo "$GITHUB_WORKFLOW_REF" | sed -E 's|^.*/(\.github/workflows/[^@]+)@\S*$|\1|')
          echo "workflow_link=https://github.com/$GITHUB_REPOSITORY/blob/$GITHUB_SHA/$workflow_path" >> "$GITHUB_ENV"
```

Follow-on suggestion: Avoid sed and a regexp substitution and do simpler fixed prefix/suffix stripping.

```suggestion
          workflow_path="$GITHUB_WORKFLOW_REF"
          workflow_path="${workflow_path#"$GITHUB_REPOSITORY"/}"
          workflow_path="${workflow_path%%@*}"
          echo "workflow_link=https://github.com/$GITHUB_REPOSITORY/blob/$GITHUB_SHA/$workflow_path" >> "$GITHUB_ENV"
```

(Either way, note that workflow filenames can contain `@`, so none of the code will work in that case, but it won't be unsafe.)